---
title: "RNA-Seq Standard Report"
output:
    html_document:
        toc: true
        toc_float: true
    
---

```{r, label="setup", include=FALSE}
library(tidyverse)
library(knitr)
```

## 1.Introduction

This report provides a comprehensive overview of the RNA-Seq standard analysis pipeline, 
detailing each step from raw data processing to final results. The analysis is performed using the `R`  and `Python` programming language and 
various bioinformatics software.

The report is structured into several sections, each corresponding to a specific step in the RNA-Seq analysis workflow. The sections include:

- **Quality Control**: This section covers the assessment of the quality of raw sequencing data using `fastp`.
- **Alignment**: This section describes the alignment of reads to a reference genome using `HISAT2`.
- **Quantification**: This section details the quantification of gene expression levels using `featureCounts`
from `Subread`.  

## 2.Quality Control

The quality of the raw sequencing data is assessed using `fastp`, which provides a comprehensive set of metrics to evaluate the quality of the reads.

The following table summarizes the quality control metrics for each sample:

```{r, label="qc_table", echo=FALSE, warning=FALSE, message=FALSE}
qc_table <- read_csv(snakemake@input[[1]])
kable(qc_table)
```

## 3.Alignment
The alignment of reads to the reference genome is performed using `HISAT2`. 

The following table summarizes the alignment metrics for each sample:
```{r, label="alignment_table", echo=FALSE, warning=FALSE, message=FALSE}
alignment_table <- read_csv(snakemake@input[[2]])
kable(alignment_table)
```

## 4.Quantification

The quantification of gene expression levels is performed using `featureCounts` from `Subread`.

The following table shows the first 10 raws of the quantification metrics for each sample:

```{r,label="quantification_table", echo=FALSE, warning=FALSE, message=FALSE}
quantification_table <- read_csv(snakemake@input[[3]])
kable(head(quantification_table))
```

## Reference

### Software and Tools

The analysis was performed using the following software and tools:

| Software | Version | Description |
|--------|-------|-----------|
| [fastp](https://github.com/OpenGene/fastp) | 0.24.0 | raw reads quality control |
| [HISAT2](https://github.com/DaehwanKimLab/hisat2) | 2.2.1 | clean reads mapping to reference genome and genome index |
| [Subread](https://github.com/ShiLab-Bioinformatics/subread) | 2.0.8 | transcription quantification: count reads to genomic features |
| [R](https://www.r-project.org/) | 4.3.3 | generate reports and data visualization |
| [Python](https://www.python.org/) | 3.13.0 | integrate shell command with snakemake and process data |